File: R/r-advanced-overview/Company.py
class Company():
    def __init__(self, HC, EBIT):
        self.HC = HC
        self.EBIT = EBIT
    def __add__(self, other):
        return Company((self.HC+other.HC)*0.8 ,(self.EBIT + other.EBIT)*1.2)
    def __repr__(self):
        out="HC: %s, EBIT: %s"%(self.HC,self.EBIT)
        return out

# a few calls:
# Microsoft = Company(50, 95)
# LinkedIn = Company(2, 5)
# Microsoft + LinkedIn 

File: R/r-advanced-overview/Company.cs
class Company
{
	private double HC;
	private double EBIT;
	
	public Company(double HC, double EBIT)
	{
		this.HC = HC;
		this.EBIT = EBIT;
	}

	public static operator +(Company A, Company B)
  	{
    		double HC = (A.HC + B.HC)*0.8;
    		double EBIT = (A.EBIT + B.EBIT)*1.2;
		return new Company(HC, EBIT)
	}
}

// a few calls:
// Company Microsoft = new Company(50, 95)
// Company LinkedIn = new Company(2, 5)
// Microsoft + LinkedIn